py/dml/dmlc.py

```diff
@@ -481,6 +481,26 @@ def show_illegal_attributes(option, opt, value, parser):
         else:
             defs[name] = value
 
+    if options.api_version not in api_versions():
+        prerr("dmlc: the version '%s' is not a valid API version" % (
+                options.api_version))
+        sys.exit(1)
+
+    if options.full_module and options.api_version not in ['4.8']:
+        prerr("dmlc: the -m option is only valid together with --api=4.8"
+              " or older")
+        sys.exit(1)
+
+    # This pair of warnings are disabled by default below Simics 7 due to sheer
+    # prominence of the issues they warn about in existing code.
+    # By only enabling the warnings in Simics 7, we allow existing codebases
+    # to handle the bugs as part of migration, instead of suddenly
+    # overwhelming them with a truly massive amount of warnings in an
+    # intermediary release.
+    if options.api_version < '7':
```

Comment:
I prefer to phrase it as `not in {'4.8', '5', '6'}`; both for subtle subjective reasons and because `'10'` < `'7'`.

Reply:
Oh yeah, true. This is not the only place in DMLC we do checks like this -- we should do a once-over.